### PR TITLE
mpool/udreg: manual patch to use mntent

### PR DESCRIPTION
**Pull request by hppritcha**
This is a hand-coded patch to udreg mpool
component to use the mntent interface to
parse /proc/mounts entries.

Needed for XC/KNL-xeons running CLE 6
or higher because, depending on use of images
at customer site, there may be some very large
entries in /proc/mounts which broke the original
parsing code, leading to failure to launch, segfaults,
etc.

@hjelmn 

Signed-off-by: Howard Pritchard howardp@lanl.gov

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/2019/ for details.

File: opal/mca/mpool/udreg/mpool_udreg_component.c [hunk 1]
```diff
@@ -147,28 +159,49 @@ static void udreg_find_hugepages (void) {
...
+
+        if (!tok) {
+#if defined(USE_STATFS)
+            struct statfs info;
+
+            statfs (mntent->mnt_dir, &info);
+#elif defined(HAVE_STATVFS)
+            struct statvfs info;
+            statvfs (mntent->mnt_dir, &info);
+            page_size = info.f_bsize;
```
Comment (hjelmn) [inline opal/mca/mpool/udreg/mpool_udreg_component.c]: Setting the page_size variable needs to happen outside the #if region.

**Comment by hjelmn** [comment]
One minor bug but otherwise good to go. Let me know when it is fixed and I will mark as reviewed.

**Comment by hppritcha** [comment]
@hjelmn I fixed it :smiley:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/2081/ for details.

**Comment by hppritcha** [comment]
@hjelmn please review again

**Comment by hjelmn** [comment]
:+1:

**Pull request merged**